src/orquestra/sdk/_ray/_build_workflow.py

```diff
@@ -152,7 +162,12 @@ def __call__(self, *wrapped_args, **wrapped_kwargs):
         for name, kwarg in wrapped_kwargs.items():
             kwargs[name] = self._unpack_argument(kwarg, name)
 
-        return self._user_fn(*args, **kwargs)
+        # if the dry_run mode is enabled, we don't execute user function, but just
+        # empty nop function returning Nones
+        if self._dry_run:
+            return self._nop_function_factory()()
+        else:
+            return self._user_fn(*args, **kwargs)
```

Comment:
**nitpick**: this class (`ArgumentUnwrapper`) is about unwrapping arguments to make it suitable for interfacing with Ray tasks. The `dry_run` is an unrelated feature. Can you extract it to a different component? Functions are composable.